agreement/pseudonode.go

```diff
@@ -386,6 +386,7 @@ func (t pseudonodeVotesTask) execute(verifier *AsyncVoteVerifier, quit chan stru
 
 	var totalWeight uint64
 	for _, result := range verifiedResults {
+		t.node.keys.Record(result.v.R.Sender, result.v.R.Round, account.Vote)
```

Comment:
consider moving this to 
```golang
case t.out <- messageEvent{T: voteVerified, Input: r.message, Err: makeSerErr(r.err)}:
```
as it would be more CPU efficient as well as more "correct" in the case where we failed to persist.